File: 10166.py
from fractions import gcd

d1,d2 = map(int,raw_input().split(' '))
temp = set()
counts = 1

for d in xrange(d1,d2+1):
  for i in xrange(1,d):
    if (str(i) + '/' + str(d)) not in temp:
      gcd_i_d = gcd(i,d)
      dd = d / gcd_i_d
      ii = i / gcd_i_d
      j = 1
      while dd*j <= d2:
        temp.add(str(ii*j) + '/' + str(dd*j))
        j += 1
      counts += 1

print counts

File: 10166.rb
d1,d2 = gets.chomp.split(' ').map{ |e| e.to_i }
temp = {}
counts = 1

d1.upto(d2) do |d|
  1.upto(d-1) do |i|
    if !temp[[i,d].to_s]
      gcd_i_d = i.gcd(d)
      dd = d / gcd_i_d
      ii = i / gcd_i_d
      j = 1
      while dd*j <= d2
        temp[[j*ii,j*dd].to_s] = true
        j += 1
      end
      counts += 1
    end
  end
end

puts counts